dashboard/app/models/contact_rollups_pardot_memory.rb

```diff
@@ -138,21 +155,26 @@ def self.update_pardot_prospects
   def self.query_new_contacts
     # New contacts are the ones exist in the production database but not in Pardot
     # (i.e., no valid Pardot IDs in contact_rollups_pardot_memory.)
-    # In addition, they must not be previously rejected by Pardot as invalid emails.
+    # In addition, they must not be previously rejected by Pardot as invalid emails
+    # or have been deleted by someone in Pardot.
     <<-SQL.squish
       SELECT processed.email, processed.data
       FROM contact_rollups_processed AS processed
       LEFT OUTER JOIN contact_rollups_pardot_memory AS pardot
         ON processed.email = pardot.email
       WHERE pardot.pardot_id IS NULL
         AND NOT (pardot.data_rejected_reason <=> '#{PardotHelpers::ERROR_INVALID_EMAIL}')
+        AND NOT (pardot.data_rejected_reason <=> '#{PardotHelpers::ERROR_PROSPECT_DELETED_FROM_PARDOT}')
```

Comment:
TIL [the NULL-safe equal operator `<=>`](https://dev.mysql.com/doc/refman/8.0/en/comparison-operators.html#operator_equal-to). :+1: